src/libraries/System.Data.Common/src/System/Data/Common/DbTransaction.cs

```diff
@@ -68,5 +68,129 @@ public virtual Task RollbackAsync(CancellationToken cancellationToken = default)
+        /// <see cref="RollbackAsync(string, System.Threading.CancellationToken)"/> and <see cref="ReleaseAsync"/> as
+        /// well as their synchronous counterparts are expected to throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        /// <remarks>
+        /// Savepoint support may vary across database versions and editions. As a result, for maximum compatibility,
+        /// always check this property on a <see cref="DbTransaction"/> instance representing an in-progress transaction
+        /// on an open <see cref="DbConnection"/>
+        /// </remarks>
+        /// <returns>
+        /// <c>true</c> if this <see cref="DbTransaction"/> instance supports database savepoints; otherwise,
+        /// <c>false</c>.
+        /// </returns>
+        public virtual bool AreSavepointsSupported => false;
+
+        /// <summary>
+        /// Creates a savepoint in the transaction. This allows all commands that are executed after the savepoint was
+        /// established to be rolled back, restoring the transaction state to what it was at the time of the savepoint.
+        /// </summary>
+        /// <param name="savepointName">The name of the savepoint to be created.</param>
+        /// <param name="cancellationToken">
+        /// An optional token to cancel the asynchronous operation.
+        /// The default value is <see cref="CancellationToken.None"/>.
+        /// </param>
+        /// <returns>A Task representing the asynchronous operation.</returns>
+        public virtual Task SaveAsync(string savepointName, CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            try
+            {
+                Save(savepointName);
+                return Task.CompletedTask;
+            }
+            catch (Exception e)
+            {
+                return Task.FromException(e);
+            }
+        }
+
+        /// <summary>
+        /// Rolls back all commands that were executed after the specified savepoint was established.
+        /// </summary>
+        /// <param name="savepointName">The name of the savepoint to roll back to.</param>
+        /// <param name="cancellationToken">
+        /// An optional token to cancel the asynchronous operation.
+        /// The default value is <see cref="CancellationToken.None"/>.
+        /// </param>
+        /// <returns>A Task representing the asynchronous operation.</returns>
+        public virtual Task RollbackAsync(string savepointName, CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            try
+            {
+                Rollback(savepointName);
+                return Task.CompletedTask;
```

Comment:
If the underlying operation is synchronous, what is the benefit of exposing it as async?

Reply:
The entire System.Data.Common (ADO.NET) surface API follows this pattern - it's a fallback mechanism for providers which don't implement async, allowing users to use them doing sync under the hood. This may not be the best practice (not sure we'd do it today), but I did it for consistently with the rest of the API surface. Providers that do support async are expected to override the async methods in any case.